### DISCUSSION TITLE
Checkbox support / Dropdowns with multiple selection at a time

### DISCUSSION THREAD

**Discussion by gauravgund** [fastapi/fastapi#8653] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import UploadFile, File, Request, Query
app = FastAPI()
@app.get('/selections')
async def selection_and(_q : str= Query("regression", enum = ["regression", "classification"]), _p : str = Query("target", enum = ['target', 'response'])):
    return({"task": _q, "target": _p})
```

### Description

Is there a way to create checkboxes based on dynamic user inputs with FastAPI and use the checkbox answers to further analyze. Also, looking for dropdowns/search bars with multiple selection to select variables. I am able to incorporate dropdowns with one selection at a time.

Already referred this: https://eugeneyan.com/writing/fastapi-html-checkbox-download/

Best Regards,

### Operating System

Windows

### FastAPI Version

0.68.0

### Python Version

3.8

**Comment by gauravgund** [comment]
@tiangolo: Please provide your inputs/example codes for this.

**Comment by ghandic** [comment]
This isn't to do with fastapi but instead to do with openapi-ui. I'm currently building out some work to add plugins to fastapi for things like this. But you're best checking out on swagger-ui or openapi-ui repos to check feasibility.

**Comment by gauravgund** [comment]
Could you share some examples if possible?

**Comment by ghandic** [comment]
https://swagger.io/docs/open-source-tools/swagger-ui/customization/plugin-api/

Fastapi just makes use of swagger ui so you'll get more support on their GitHub repo if you need it. Could you close out this issue as it is not to do with fastapi

**Comment by Sagaryal** [comment]
> https://swagger.io/docs/open-source-tools/swagger-ui/customization/plugin-api/
> 
> Fastapi just makes use of swagger ui so you'll get more support on their GitHub repo if you need it. Could you close out this issue as it is not to do with fastapi

Could you please elaborate how to integrate it in FastAPI? Swagger UI supports it as show [here.](https://editor.swagger.io/) `/pet/findByStatus`

**Comment by Sagaryal** [comment, marked as answer]
@gauravgund I tried as following for multiple selection. And is working. Swagger-ui supports it as I mention in above comment. Swagger json requires 'array' type. For FastApi to generate 'array' type, we need to supply Array and for this case List/list worked.

```python
from enum import Enum
from typing import List
from fastapi import Query

class MyEnums(str, Enum):
    item_1 = "item_1"
    item_2 = "item_2"
    item_3 = "item_3"

@api_routes.post('/single-multi-selection')
def singleMultiSelectionField(singleSelctionDropdown: MyEnums, multiSelctionDropdown: List[MyEnums] = Query(...)):
    singleDropdownValue = singleSelctionDropdown.value
    multiSelectionList = [item.value for item in multiSelctionDropdown]
```

Also you can check this [issue](https://github.com/tiangolo/fastapi/issues/246). It provides more info on dropdowns.

**Comment by romcab** [comment]
Anyone here have issues running  List query params in aws-lambda? The code works as expected in my local machine. However, only the last item were returned when running on aws lambda. The request url correctly transform the query param to /multiselect?items=item_1&items=item_2. Below is my sample code. 

Setup:
ALB+Lambda
fastapi v0.89.0
mangum v0.17.0

Returns => ["item_2"] Assuming i selected item_1 and item_2

```
from enum import Enum
from typing import List

from fastapi import FastAPI, Query
from mangum import Mangum


class MyEmums(str, Enum):
    item_1 = "item_1"
    item_2 = "item_2"
    item_3 = "item_3"

app = FastAPI(title="My App")


@app.get("/multiselect")
def multiselect(items: List[MyEmums] = Query(...)):
    return items

handler = Mangum(app)
```

**Comment by ghandic** [comment]
Might be worth raising this on Mangum repo 😄
